src/Features/LanguageServer/Protocol/Handler/Configuration/DidChangeConfigurationNotificationHandler_OptionList.cs

```diff
@@ -0,0 +1,25 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Immutable;
+using Microsoft.CodeAnalysis.Completion;
+using Microsoft.CodeAnalysis.ImplementType;
+using Microsoft.CodeAnalysis.Options;
+using Microsoft.CodeAnalysis.SymbolSearch;
+
+namespace Microsoft.CodeAnalysis.LanguageServer.Handler.Configuration
+{
+    internal partial class DidChangeConfigurationNotificationHandler
+    {
+        // TODO: all the supported options here
+        private static readonly ImmutableArray<ISingleValuedOption> s_supportedGlobalOptions = ImmutableArray.Create<ISingleValuedOption>();
```

Comment:
Why do we need "supported" options? Why not just allow any options?

Reply:
Do you mean here we don't care if the options is meaningful to the cilent not and just asking it for all the options?
If case the client doesn't have that, (it might return null), we just ignore the value and fall back to our default value?

Reply:
pretty much